addons/hw_drivers/websocket_client.py

```diff
@@ -13,6 +19,50 @@
 _logger = logging.getLogger(__name__)
 websocket.enableTrace(True, level=logging.getLevelName(_logger.getEffectiveLevel()))
 
+WS_REQUEST_POOL = concurrent.futures.ThreadPoolExecutor(8)
+
+WS_REQUEST_TIME_TIMEOUT = 5 * 60  # seconds
+WS_REQUEST_COUNTER = itertools.count()
+"""
+unique incrementing identifier each time a new websocket message is received
+"""
+
+def _check_ws_request_pool():
+    """Check if the request pool is not stuck with timeout-ing requests.
+    We can't kill a thread, but if the situation is desperate we will restart the IoT.
+
+    We use a concurrent.futures.ThreadPoolExecutor to handle the websocket messages.
+    Ccenario example:
+     When a ws request R1 is received, it will create a thread T1 in the pool to handle the request
+     When a second ws request is received R2:
+      if R1 finished, T1 is "free" so it will be reused
+      if R1 is still running, T1 will be busy so a second thread T2 will be created to handle R2
+     ...
+     until the maximum number of threads is reached (see `ThreadPoolExecutor._max_workers`).
+     In this case they will wait until a thread is free.
+    """
+    def thread_stack_trace(thread):
+        frame = sys._current_frames().get(thread.ident, None)
+        if frame:
+            return ". Thread stack:\n" + ''.join(traceback.format_stack(frame))
+        return ''
```

Comment:
can be replaced by `misc.dumpstacks` ? we can give the thread_idents to it and change the used parameter to match with the intended one (if it applies)